File: src/DigitalSignage.Infrastructure/Models/Settings/Display.cs
// SPDX-FileCopyrightText: © 2014 Oberverwaltungsgericht Rheinland-Pfalz <poststelle@ovg.jm.rlp.de>
// SPDX-License-Identifier: EUPL-1.2
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DigitalSignage.Infrastructure.Models.Settings;

[Table("Displays")]
public class Display
{
    [Key]
    public int Id { get; set; }

    public string Name { get; set; }
    public string Title { get; set; }
    public string Template { get; set; }
    public string Styles { get; set; }
    public string Filter { get; set; }
    public string Group { get; set; }
    public string ControlUrl { get; set; }
    public string NetAddress { get; set; }
    public string WolIpAddress { get; set; }
    public string WolMacAddress { get; set; }
    public int WolUdpPort { get; set; }
    public string Description { get; set; }
    public ICollection<NoteAssignment> NotesAssignments { get; set; }
    public bool Dummy { get; set; }
}

File: src/ds-webapps/src/models/Display.ts
// SPDX-FileCopyrightText: © 2014 Oberverwaltungsgericht Rheinland-Pfalz <poststelle@ovg.jm.rlp.de>
// SPDX-License-Identifier: EUPL-1.2
export default interface Display {
  id: number
  name: string
  title: string
  template: string
  styles: string
  filter: string
  group: string
  controlUrl: string
  netAddress: string
  wolIpAddress: string
  wolMacAddress: string
  wolUdpPort: number
  description: string
  dummy: boolean
}
